File: java_backend/src/main/java/com/example/webrtccall/dto/SignalMessage.java
package com.example.webrtccall.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SignalMessage {
    private String type; // offer, answer, ice-candidate, join-room, leave-room
    private String roomId;
    private String userId;
    private String targetUserId;
    private Object data; // SDP or ICE candidate data
}

File: Android/app/src/main/java/com/example/webrtc_call_android/data/model/SignalMessage.kt
package com.example.webrtc_call_android.data.model

data class SignalMessage(
    val type: String, // offer, answer, ice-candidate, join-room, leave-room
    val roomId: String? = null,
    val userId: String? = null,
    val targetUserId: String? = null,
    val data: Any? = null
)

